src/servant/tt_001/F84.c

```diff
@@ -152,7 +156,89 @@ Entity* func_us_80171568(Entity* self, s32 entityId) {
     // BUG? There is a fall-through case here with no return value on PSX
 }
 
-INCLUDE_ASM("servant/tt_001/nonmatchings/F84", func_us_80171624);
+void func_us_80171624(s32 arg0) {
+    RECT rect;
+    u16* dst;
+    u16* src;
+    s32 i;
+    SpriteParts** spriteBanks;
+    Entity* e;
+    u16 temp;
+
+#ifdef VERSION_PC
+    // i exceeds the size of D_80170448
+    const int len = LEN(D_80170448);
+#else
+    const int len = 256;
+#endif
+
+    if (arg0 != 3) {
+        dst = &g_Clut[0x1400];
+        src = D_us_80170448;
+
+        for (i = 0; i < len; i++) {
+            temp = *src++;
+            *dst = temp;
+            if (i & 0xF) {
+                *dst = temp | 0x8000;
+            }
+            dst++;
+        }
+
+        dst = &g_Clut[0x1430];
+        src = D_us_80170580;
+
+        for (i = 0; i < 32; i++) {
+            *dst++ = *src++;
+        }
+
+        rect.x = 0;
+        rect.w = 0x100;
+        rect.h = 1;
+        rect.y = 0xF4;
+
+        dst = &g_Clut[0x1400];
+        LoadImage(&rect, dst);
+
+        spriteBanks = g_api.o.spriteBanks;
+        spriteBanks += 20;
+        *spriteBanks = (SpriteParts*)D_us_80170040;
+
+        e = &g_Entities[4];
+        DestroyEntity(e);
+        e->unk5A = 0x6C;
+        e->palette = 0x140;
+        e->animSet = ANIMSET_OVL(20);
+        e->entityId = 0xD1;
+        e->params = 0;
+        e->zPriority = PLAYER.zPriority - 2;
+        e->facingLeft = (PLAYER.facingLeft + 1) & 1;
+        e->posX.val = PLAYER.posX.val;
+        e->posY.val = PLAYER.posY.val;
+
+        if (arg0 == 1) {
+            e->entityId = 0xD1;
```

Comment:
Seems like this is creating an entity within this overlay https://github.com/Xeeynamo/sotn-decomp/blob/cf280c66f5ce0463834aaacf1dc56f4991979d80/src/dra/7A4D0.c#L100

D_80170000 is supposed to point to ServantDesc, so this will create an entity with func_us_80171864 as the update func I think? Assuming I'm correct it would be good to document this

Reply:
I agree.  Also, the bat shares the same entityId as well, which means it's update function is func_80172120.  I was planning on doing some refactoring with things that are the same between tt_000 and tt_001 for my next task.  I'll document this find with that.

This looks like it just fetches the update though.  I don't think it actually creates the entity.  

I think the entity gets created here
https://github.com/Xeeynamo/sotn-decomp/blob/cf280c66f5ce0463834aaacf1dc56f4991979d80/src/dra/loading.c#L130

Which gets called from the Menu and from another part of the game engine.